src/test/java/io/supertokens/test/oauth/api/OAuthAPITest.java

```diff
@@ -0,0 +1,102 @@
+import io.supertokens.pluginInterface.STORAGE_TYPE;
+import io.supertokens.pluginInterface.exceptions.InvalidConfigException;
+import io.supertokens.pluginInterface.exceptions.StorageQueryException;
+import io.supertokens.pluginInterface.multitenancy.exceptions.TenantOrAppNotFoundException;
+import io.supertokens.pluginInterface.oauth.OAuthAuthResponse;
+import io.supertokens.pluginInterface.oauth.OAuthStorage;
+import io.supertokens.storageLayer.StorageLayer;
+import io.supertokens.test.TestingProcessManager;
+import io.supertokens.test.Utils;
+import io.supertokens.webserver.InputParser;
+import org.junit.*;
+import org.junit.rules.TestRule;
+
+import java.io.IOException;
+
+import static org.junit.Assert.assertNotNull;
+
+public class OAuthAPITest {
+    TestingProcessManager.TestingProcess process;
+
+    @Rule
+    public TestRule watchman = Utils.getOnFailure();
+
+    @AfterClass
+    public static void afterTesting() {
+        Utils.afterTesting();
+    }
+
+    @Before
+    public void beforeEach() throws InterruptedException {
+        Utils.reset();
+        String[] args = {"../"};
+
+        this.process = TestingProcessManager.start(args);
+        FeatureFlagTestContent.getInstance(process.getProcess())
+                .setKeyValue(FeatureFlagTestContent.ENABLED_FEATURES, new EE_FEATURES[]{EE_FEATURES.OAUTH});
+        process.startProcess();
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STARTED));
+
+        if (StorageLayer.getStorage(process.getProcess()).getType() != STORAGE_TYPE.SQL) { //TODO check if this is true here also
+            return;
+        }
+    }
+
+    @After
+    public void afterEach() throws InterruptedException {
+        process.kill();
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STOPPED));
+    }
+
+    @Test
+    public void test()
+            throws StorageQueryException, OAuthException, HttpResponseException, TenantOrAppNotFoundException,
+            InvalidConfigException, IOException {
+
+        String clientId = "a685663d-1b5d-4a70-b7f7-025ff2e2d7a4";
+        String redirectUri = "http://localhost.com:3031/auth/callback/ory";
+        String responseType = "code";
+        String scope = "profile";
+        String state = "%EF%BF%BD%EF%BF%BD%EF%BF%BD%EF%BF%BD%EF%BF%BD%EF%BF%BDv%EF%BF%BD%EF%BF%BD%EF%BF%BD%EF%BF%BD%EF%BF%BD";
+
+        OAuthStorage oAuthStorage = (OAuthStorage) StorageLayer.getStorage(process.getProcess());
+
+        OAuthAuthResponse response = OAuth.getAuthorizationUrl(process.getProcess(), null, oAuthStorage,  clientId, redirectUri, responseType, scope, state);
+
+        System.out.println(response);
+        System.out.println(response.redirectTo);
+
+        for(String cooke : response.cookies){
+            System.out.println(cooke);
+        }
+
+        assertNotNull(response);
+        assertNotNull(response.redirectTo);
+        assertNotNull(response.cookies);
```

Comment:
need to assert actual values and not just that they are not null

Reply:
Will do when actually working on the tests, not just using them as a trigger point